### PR TITLE
docs: synthetic view

### LINKED ISSUE #1117: Element enhancing guide
Would be great to have a guide on how to dynamically enhance elements.
In Au1 we had `TemplatingEngine.enhance`. I managed to hack together some code which kind of works
```ts
    const renderPlan = createElement(this.platform, MdcDialog, props);
    const sv = renderPlan.createView(this.container);
    const dialogEl = sv.children![0].host!;
    sv.activate(sv, this.appRoot.controller, LifecycleFlags.none, Scope.create(bindingContext));
    document.body.appendChild(dialogEl);
```
In my opinion, this is too verbose and unreliable. 
On top of that, passing `props` to set element attributes fails, see https://gist.dumber.app/?gist=29f1d7ef8a1d002261dd1d93e0973183

As a part of the guide, would be also great to demonstrate how to project another element into the enhanced element `au-slot`.
This is a must have feature for MDC bridge or Aurelia UX dialogs.

### PR DESCRIPTION
**Pull request by Sayan751**
# Pull Request

### 🎫 Issues

Closes #1117

### CODE REVIEW

File: docs/user-docs/tutorials/synthetic-view.md [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    // Step#5: Create and activate view
+    const view = this.view = factory
+      .create(controller)
+      .setLocation(loc); //<-- render location created in Step#2
+
+    // binds the view with the given scope and attaches it to the DOM
+    await view.activate(
+      view,
+      controller,
+      LifecycleFlags.None,
```
Comment (zewa666) [inline docs/user-docs/tutorials/synthetic-view.md:357]: shouldn't this be LifecycleFlags.none?

File: docs/user-docs/tutorials/synthetic-view.md [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+
+{% hint style="success" %}
+You can see a working example [here](https://stackblitz.com/edit/typescript-hh9a1e).
+{% endhint %}
+
+## Prerequisites
+
+Before going any further, you should be familiar with some basic Aurelia concepts as well as some fundamental Javascript ones as well. While these are not hard prerequisites, please know that some concepts used in this tutorial out of context might be confusing or difficult to understand.
+
+* Creating a new Aurelia app. THis won't be covered in this tutorial and you may refer the other tutorials to this end.
```
Comment (zewa666) [inline docs/user-docs/tutorials/synthetic-view.md:24]: T(H)is won't be ...

File: docs/user-docs/tutorials/synthetic-view.md (hunk 1 above)
Comment (Sayan751) [reply docs/user-docs/tutorials/synthetic-view.md]: Done. Thank you for the review.

File: docs/user-docs/tutorials/synthetic-view.md (hunk 2 above)
Comment (Sayan751) [reply docs/user-docs/tutorials/synthetic-view.md]: Done. Thank you for the review.

File: docs/user-docs/tutorials/synthetic-view.md [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+- Then we will generate a template that is constructed on runtime using these custom elements.
+- Finally we will use lower level Aurelia APIs such as `ViewFactory` and `Controller` to hydrate the template.
+
+{% hint style="success" %}
+You can see a working example [here](https://stackblitz.com/edit/typescript-hh9a1e).
+{% endhint %}
+
+## Prerequisites
+
+Before going any further, you should be familiar with some basic Aurelia concepts as well as some fundamental Javascript ones as well. While these are not hard prerequisites, please know that some concepts used in this tutorial out of context might be confusing or difficult to understand.
```
Comment (bigopon) [inline docs/user-docs/tutorials/synthetic-view.md:22]: 👍 x2

File: docs/user-docs/tutorials/synthetic-view.md [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+export class App implements ICustomElementViewModel {
+  private view: ISyntheticView | undefined;
+  private readonly bc = { value: 'Hello Aurelia!' };
+  //...
+
+  public async add() {
+    //...
+    // Step#5: Create and activate view
+    const view = this.view = factory
+      .create(controller)
```
Comment (bigopon) [inline docs/user-docs/tutorials/synthetic-view.md:350]: while refactoring some of the factory/node sequence APIs, I kind of looked into having loc as 2nd parameter of this create method too. Though there' some difficulties related to timing so it hasn't been done yet. This reminds me of that, having that should allow us save soeme more KB in the code and example too.

File: docs/user-docs/tutorials/synthetic-view.md [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+}
+
+```
+{% endtab %}
+{% endtabs %}
+
+## Live example
+
+You can see the live example below.
+It involves some trivial guard conditions; other than that it is mostly the same.
```
Comment (bigopon) [inline docs/user-docs/tutorials/synthetic-view.md:424]: 👍

**Review by bigopon — APPROVED** [review]
So much work in this documentation, awesome effort, thanks @Sayan751 👍

**Comment by bigopon** [comment]
@Sayan751 can you help resolve the conflict in the TOC file?

**Comment by Sayan751** [comment]
> @Sayan751 can you help resolve the conflict in the TOC file?

@bigopon  done

**Pull request merged**